core/controllers/base_test.py

```diff
@@ -234,16 +234,6 @@ def test_html_requests_have_no_store_cache_policy(self):
             'must-revalidate, no-cache, no-store'
         )
 
-    def test_root_redirect_rules_for_logged_in_learners(self):
```

Comment:
Where are the new tests for these?

Reply:
Earlier the access validation and the subsequent action based on the result of the validation was done by the backend.
But now, the code for access_validation is present at the backend (`access_validation.py`), and code for subsequent action is placed in the frontend (route guards and internal validation). Both parts have their own unit tests (`access_validation_test.py` and frontend unit tests for route guards and page-root components`. So, the behaviour which was tested before cannot be tested in the backend as half of the functionality is moved to the frontend. But the subparts can be tested on the backend and frontend separately.

Example:
Earlier, there was a test in the backend which tests the behavior that the user is redirected to the preferred dashboard page if he is logged in and tries to reach the splash page.

Now, it is tested in the backend that access validation provides the preferred dashboard as JSON response and it is tested in the frontend that user is redirected to the preferred dashboard which is provided by the access validation handler.
So, the combination of these two tests does end up testing the same behaviour which was tested earlier in the backend.

Reply:
OK, this sounds good. Thanks for explaining!